float32 tactor1
float32 tactor2
float32 skin_stretch
